lib/ecto/adapters/myxql.ex

```diff
@@ -312,27 +312,42 @@ defmodule Ecto.Adapters.MyXQL do
   def structure_dump(default, config) do
     table = config[:migration_source] || "schema_migrations"
     path  = config[:dump_path] || Path.join(default, "structure.sql")
+    dump_prefixes = config[:dump_prefixes] || [config[:database]]
 
-    with {:ok, versions} <- select_versions(table, config),
-         {:ok, contents} <- mysql_dump(config),
+    with {:ok, versions} <- select_versions(dump_prefixes, table, config),
+         {:ok, contents} <- mysql_dump(dump_prefixes, config),
```

Comment:
```suggestion
    with {:ok, versions} <- select_versions(prefixes, table, config),
         {:ok, contents} <- mysql_dump(prefixes, config),
```
to be consistent with postgres and also because we are already inside the dump functions